### PR TITLE
Change NggCompactSubgroup mode to NggCompactDisable

### PR DESCRIPTION
**Pull request by amdrexu**
Subgroup compaction is deprecated since its performance gain is not
satisfying. Thus, vertex compaction is the default compaction mode.
In the future, we will support "compactionless" mode that is to
disable compaction and relevant vertex mapping to save LDS read/write
operations. Here, we remove old enumerant "NggCompactSubgroup" and
add a new one "NggCompactDisable".

Change-Id: Ife7b0e6ea279ee69c7e0c99685733a1996cea55c

### CODE REVIEW

File: llpc/include/llpc.h [hunk 1]
```diff
@@ -39,37 +39,37 @@ using Vkgc::BinaryData;
...
 using Vkgc::GraphicsPipelineBuildInfo;
-using Vkgc::OutputAllocFunc;
-using Vkgc::PipelineOptions;
-using Vkgc::Result;
-using Vkgc::ShaderModuleData;
-using Vkgc::ShaderModuleEntryData;
-using Vkgc::DescriptorRangeValue;
 using Vkgc::MaxColorTargets;
-using Vkgc::NggCompactSubgroup;
+using Vkgc::NggCompactDisable;
```
Comment (amdrexu) [inline llpc/include/llpc.h:47]: The only real change is here. Others are for clang-format.

**Comment by amdvlk-admin** [comment]
### Test summary for commit 216b261d60e10dc16d071a10b4757b6a67f4ec8f
<details open><summary>Driver commits used in build</summary><ul>

<li>CWPACK: amd-master 7387247eb9889ddcabbc1053b9c2052e253b088e</li>
<li>METROHASH: amd-master 712f76fee75d69b23a1ea8f6465752c3ccaaf9a2</li>
<li>PAL: dev 82390674c40402c81dbf311a5ee488fb972894f5</li>
<li>SPVGEN: dev 2f679769a8491f423f5cf2f06adfa2870a82935d</li>
<li>XGL: dev 116d28a446eba3a9f8e6f577bab34fc07e090993</li>
<li>LLVM-PROJECT: amd-gfx-gpuopen-dev a7cfa00580cd7aa8e66f254d1e200e976d90b08e</li>
</ul></details>

<details open><summary>ShaderDB test</summary><ul>

<li>Passed: 962/962 (100.0%)</li>
<li>Failed: 0/962 (0.0%)</li>
</ul></details>

<details open><summary>CTS tests (built with branch Vulkan-CTS-1.2.0.2-fetch-sources-fix)</summary><ul>


<details open><summary>Ubuntu 18.04, Gfx8</summary><ul>

<li>Passed: 103142/103143 (100.0%)</li>
<li><details><summary>Failed: 1/103143 (0.0%)</summary><p>


Failures:
```
FAILURE: dEQP-VK.api.object_management.multithreaded_shared_resources.graphics_pipeline
Stack trace: result: 0xfffffffffffffff3 at vktApiObjectManagementTests.cpp:1827
```
</p></details></li>


</ul></details>

<details open><summary>Rhel 7.6, Gfx9</summary><ul>

<li>Passed: 103199/103199 (100.0%)</li>
<li>Failed: 0/103199 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 18.04, Gfx10</summary><ul>

<li>Passed: 103199/103199 (100.0%)</li>
<li>Failed: 0/103199 (0.0%)</li>
</ul></details>


</ul></details>

**Comment by JaxLinAMD** [comment]
retest this please

**Comment by amdvlk-admin** [comment]
### Test summary for commit 216b261d60e10dc16d071a10b4757b6a67f4ec8f
<details open><summary>Driver commits used in build</summary><ul>

<li>CWPACK: amd-master 7387247eb9889ddcabbc1053b9c2052e253b088e</li>
<li>METROHASH: amd-master 712f76fee75d69b23a1ea8f6465752c3ccaaf9a2</li>
<li>PAL: dev 82390674c40402c81dbf311a5ee488fb972894f5</li>
<li>SPVGEN: dev 2f679769a8491f423f5cf2f06adfa2870a82935d</li>
<li>XGL: dev 116d28a446eba3a9f8e6f577bab34fc07e090993</li>
<li>LLVM-PROJECT: amd-gfx-gpuopen-dev a7cfa00580cd7aa8e66f254d1e200e976d90b08e</li>
</ul></details>

<details open><summary>ShaderDB test</summary><ul>

<li>Passed: 962/962 (100.0%)</li>
<li>Failed: 0/962 (0.0%)</li>
</ul></details>

<details open><summary>CTS tests (built with branch Vulkan-CTS-1.2.0.2-fetch-sources-fix)</summary><ul>


<details open><summary>Ubuntu 18.04, Gfx8</summary><ul>

<li>Passed: 103143/103143 (100.0%)</li>
<li>Failed: 0/103143 (0.0%)</li>
</ul></details>

<details open><summary>Rhel 7.6, Gfx9</summary><ul>

<li>Passed: 103199/103199 (100.0%)</li>
<li>Failed: 0/103199 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 18.04, Gfx10</summary><ul>

<li>Passed: 103199/103199 (100.0%)</li>
<li>Failed: 0/103199 (0.0%)</li>
</ul></details>


</ul></details>

**Pull request merged**